elrond-wasm/src/api/send_api.rs

```diff
@@ -131,6 +106,46 @@ pub trait SendApi: ErrorApi + Clone + Sized {
 		)
 	}
 
+	/// Performs a simple ESDT/NFT transfer, but via async call.  
+	/// As with any async call, this immediately terminates the execution of the current call.  
+	/// So only use as the last call in your endpoint.  
+	/// If you want to perform multiple transfers, use `self.send().direct()` instead.  
+	/// Note that EGLD can NOT be transfered with this function.  
```

Comment:
so right new we have direct and transfer_via_async_call combined - can we do the same for executeOnDestination contract with token transfer ?

Reply:
executeOnDest context is a bit of a special case. There is a separate way of doing this: Andrei Marinica created a way of importing a contract's proxy, which lets the "parent" contract be able to call endpoints as if they were simple functions. This proxy automatically handles #[payment] arguments and performs the needed contract call.

TL;DR: These functions are not used directly.